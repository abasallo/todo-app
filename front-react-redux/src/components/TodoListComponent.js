import React from 'react'
import { Component, PropTypes } from 'react'
import { List, ListItem, Checkbox, IconButton } from 'material-ui'
import ActionGrade from 'material-ui/svg-icons/action/delete'

import config from '../config'

class TodoListComponent extends Component {

    render() { return (
        <List style={ config.todoListStyle }>
            { this.props.todos.map(todo =>
                <ListItem key={ todo.id } primaryText={ todo.text }
                          leftCheckbox={ <Checkbox checked={ todo.completed } onClick={ () => this.props.onToggleClick(todo.id) }/>}
                          rightIcon={ <IconButton touch={ true } onClick={ () => this.props.onDeleteClick(todo.id) } style={ config.todoListDeleteButtonStyle }>
                              <ActionGrade/>
                          </IconButton > }
                />
            )}
        </List> )}
}

TodoListComponent.propTypes = {
    todos: PropTypes.array.isRequired,
    onToggleClick: PropTypes.func.isRequired,
    onDeleteClick: PropTypes.func.isRequired }

export default TodoListComponent
