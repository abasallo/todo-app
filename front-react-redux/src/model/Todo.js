import FilterTypes from '../../src/model/FilterTypes'

import config from '../config'

export default class Todo {

    static filterTodoArrayBy = (todos, filter) => {

        let todosCopy = todos.slice()
        switch (filter) {
            case FilterTypes.showAll:
                return todosCopy
            case FilterTypes.showCompleted:
                return todosCopy.filter(todo => todo.completed)
            case FilterTypes.showActive:
                return todosCopy.filter(todo => !todo.completed)
            default:
                throw new Error('Unknown filter: ' + filter)
        }
    }

    constructor(text = config.todo.defaults.text, completed = false, id = -1) {
        this.id = id
        this.text = text
        this.completed = completed
    }

    toggle() {
        this.completed = !this.completed
        return this
    }
}
