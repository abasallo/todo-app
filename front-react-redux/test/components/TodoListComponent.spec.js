import React from 'react'
import { shallow } from 'enzyme'

import Todo from '../../src/model/Todo'
import TodoListComponent from '../../src/components/TodoListComponent'

it('renders properly', () => {

    const firstTodo = new Todo('Task 1', true)
    const secondTodo = new Todo('Task 2', false)
    const todos = [ firstTodo, secondTodo ]

    expect(shallow(<TodoListComponent todos={todos} onToggleClick={ (id) => {} } onDeleteClick={ (id) => {} }/>)).toMatchSnapshot()
})
