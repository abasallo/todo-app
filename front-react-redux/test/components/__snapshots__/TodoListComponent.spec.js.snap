exports[`test renders properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <List
    style={undefined}>
    <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        insetChildren={false}
        leftCheckbox={
            <Checkbox
                checked={true}
                disabled={false}
                labelPosition="right"
                onClick={[Function]} />
        }
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Task 1"
        primaryTogglesNestedList={false}
        rightIcon={
            <IconButton
                disableTouchRipple={false}
                disabled={false}
                iconStyle={Object {}}
                onClick={[Function]}
                style={undefined}
                tooltipPosition="bottom-center"
                touch={true}>
                <ActionDelete />
            </IconButton>
        }
        secondaryTextLines={1} />
    <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        insetChildren={false}
        leftCheckbox={
            <Checkbox
                checked={false}
                disabled={false}
                labelPosition="right"
                onClick={[Function]} />
        }
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Task 2"
        primaryTogglesNestedList={false}
        rightIcon={
            <IconButton
                disableTouchRipple={false}
                disabled={false}
                iconStyle={Object {}}
                onClick={[Function]}
                style={undefined}
                tooltipPosition="bottom-center"
                touch={true}>
                <ActionDelete />
            </IconButton>
        }
        secondaryTextLines={1} />
</List>,
  "nodes": Array [
    <List
      style={undefined}>
      <ListItem
            autoGenerateNestedIndicator={true}
            containerElement="span"
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftCheckbox={
                  <Checkbox
                        checked={true}
                        disabled={false}
                        labelPosition="right"
                        onClick={[Function]} />
            }
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Task 1"
            primaryTogglesNestedList={false}
            rightIcon={
                  <IconButton
                        disableTouchRipple={false}
                        disabled={false}
                        iconStyle={Object {}}
                        onClick={[Function]}
                        style={undefined}
                        tooltipPosition="bottom-center"
                        touch={true}>
                        <ActionDelete />
                  </IconButton>
            }
            secondaryTextLines={1} />
      <ListItem
            autoGenerateNestedIndicator={true}
            containerElement="span"
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftCheckbox={
                  <Checkbox
                        checked={false}
                        disabled={false}
                        labelPosition="right"
                        onClick={[Function]} />
            }
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Task 2"
            primaryTogglesNestedList={false}
            rightIcon={
                  <IconButton
                        disableTouchRipple={false}
                        disabled={false}
                        iconStyle={Object {}}
                        onClick={[Function]}
                        style={undefined}
                        tooltipPosition="bottom-center"
                        touch={true}>
                        <ActionDelete />
                  </IconButton>
            }
            secondaryTextLines={1} />
</List>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TodoListComponent
        onDeleteClick={[Function]}
        onToggleClick={[Function]}
        todos={
                Array [
                        Todo {
                          "completed": true,
                          "id": -1,
                          "text": "Task 1",
                        },
                        Todo {
                          "completed": false,
                          "id": -1,
                          "text": "Task 2",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodoListComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onDeleteClick": [Function],
          "onToggleClick": [Function],
          "todos": Array [
            Todo {
              "completed": true,
              "id": -1,
              "text": "Task 1",
            },
            Todo {
              "completed": false,
              "id": -1,
              "text": "Task 2",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <List
          style={undefined}>
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    leftCheckbox={
                              <Checkbox
                                        checked={true}
                                        disabled={false}
                                        labelPosition="right"
                                        onClick={[Function]} />
                    }
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Task 1"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        onClick={[Function]}
                                        style={undefined}
                                        tooltipPosition="bottom-center"
                                        touch={true}>
                                        <ActionDelete />
                              </IconButton>
                    }
                    secondaryTextLines={1} />
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    leftCheckbox={
                              <Checkbox
                                        checked={false}
                                        disabled={false}
                                        labelPosition="right"
                                        onClick={[Function]} />
                    }
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Task 2"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        onClick={[Function]}
                                        style={undefined}
                                        tooltipPosition="bottom-center"
                                        touch={true}>
                                        <ActionDelete />
                              </IconButton>
                    }
                    secondaryTextLines={1} />
</List>,
        "_debugID": 2,
        "_renderedOutput": <List
          style={undefined}>
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    leftCheckbox={
                              <Checkbox
                                        checked={true}
                                        disabled={false}
                                        labelPosition="right"
                                        onClick={[Function]} />
                    }
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Task 1"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        onClick={[Function]}
                                        style={undefined}
                                        tooltipPosition="bottom-center"
                                        touch={true}>
                                        <ActionDelete />
                              </IconButton>
                    }
                    secondaryTextLines={1} />
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    leftCheckbox={
                              <Checkbox
                                        checked={false}
                                        disabled={false}
                                        labelPosition="right"
                                        onClick={[Function]} />
                    }
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Task 2"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <IconButton
                                        disableTouchRipple={false}
                                        disabled={false}
                                        iconStyle={Object {}}
                                        onClick={[Function]}
                                        style={undefined}
                                        tooltipPosition="bottom-center"
                                        touch={true}>
                                        <ActionDelete />
                              </IconButton>
                    }
                    secondaryTextLines={1} />
</List>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoListComponent
    onDeleteClick={[Function]}
    onToggleClick={[Function]}
    todos={
        Array [
            Todo {
              "completed": true,
              "id": -1,
              "text": "Task 1",
            },
            Todo {
              "completed": false,
              "id": -1,
              "text": "Task 2",
            },
          ]
    } />,
}
`;
